maxim/tests/test_add_dataset_entries.py

```
@@ -0,0 +1,348 @@
+        """Test that add_dataset_entries raises Exception for invalid entry types."""
+        with self.assertRaises(Exception) as context:
+            self.maxim.maxim_api.add_dataset_entries(
+                os.environ["MAXIM_DATASET_ID"],
+                ["invalid_entry_type"],  # String instead of dict or DatasetEntry
+            )
+        
+        self.assertIn("Invalid entry type", str(context.exception))
+
+    def test_add_dataset_entries_invalid_dict_structure(self) -> None:
+        """Test that add_dataset_entries handles invalid dictionary structure properly."""
+        with self.assertRaises(Exception) as context:
```

Ruff (0.12.2)

Use `pytest.raises` instead of unittest-style `assertRaises`

Replace `assertRaises` with `pytest.raises`

(PT027)